### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by mhoppal**

### CODE REVIEW

File: smoke-test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+func initializeMonascaClient(token, monascaURL string)(){
+	timeoutSetting := os.Getenv("TIMEOUT")
+	var timeout int
+	if timeoutSetting == "" {
+		timeout = 5
+	} else {
+		var err error
+		timeout, err = strconv.Atoi(timeoutSetting)
+		if err != nil {
+			timeout = 5
```
Comment (craigbr) [inline smoke-test.go]: You are ignoring the configured value without telling the user

File: smoke-test.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/monasca/golang-monascaclient/monascaclient"
+	"strconv"
+	"github.com/monasca/golang-monascaclient/monascaclient/models"
+)
+
+var webhookTriggered bool = false
+var testsSucceeded int = 0
+var totalTests int = 6
```
Comment (craigbr) [inline smoke-test.go]: I wonder if it makes more sense to increment totalTests inside every test. That way adding a new test doesn't require updating this value

File: smoke-test.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+func testWebhookTrigger() {
+	// Wait 5 minutes for webhook to trigger
+	for i := 0; i < 20; i++ {
+		if webhookTriggered {
+			fmt.Println("SUCCESS")
+			testsSucceeded++
+			return
+		}
+		time.Sleep(15 * time.Second)
```
Comment (craigbr) [inline smoke-test.go]: I would wait less the 15 seconds. It is likely to happen within a second.

File: smoke-test.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+		}
+	}
+	if cleanupStatus {
+		testsSucceeded++
+		fmt.Println("SUCCESS")
+	}
+}
+
+func cleanupPreviousRun() {
+	notificationName := "smoke_test_notification"
```
Comment (craigbr) [inline smoke-test.go]: 'smoke_test_notification' is defined twice. Should be a const

File: smoke-test.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST WEBHOOK TRIGGERED")
+	testWebhookTrigger()
+	fmt.Println()
+	testsRun++
+
+	fmt.Println("TEST CLEANUP")
+	cleanup(alarmDefinitionID, notificationID)
```
Comment (craigbr) [inline smoke-test.go]: Personally, I think you should only cleanup if everything else succeeded. Doing it always makes it harder to debug issues.

File: smoke-test.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+			}
+		}
+		if notificationID != "" {
+			err := monascaclient.DeleteNotificationMethod(notificationID)
+			if err != nil {
+				fmt.Printf("Error deleting notification method - %s\n", err.Error())
+			}
+		}
+	}
+	alarmDefinitionName := "smoke_test_alarm"
```
Comment (craigbr) [inline smoke-test.go]: Should be const

File: smoke-test.go [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+	alarmDefinitions, err := monascaclient.GetAlarmDefinitions(&models.AlarmDefinitionQuery{Name: &alarmDefinitionName})
+	if err != nil {
+		fmt.Printf("Error getting alarm definitions to delete potential left over alarm definition "+
+			"from previous runs - %s\n", err.Error())
+	} else {
+		alarmDefinitionID := ""
```
Comment (craigbr) [inline smoke-test.go:188]: There should be 0 or 1 elements. Nothing to do if 0, if 1 delete it. Would be simpler

File: smoke-test.go (hunk 5 above)
Comment (mhoppal) [reply smoke-test.go]: done

File: smoke-test.go (hunk 6 above)
Comment (mhoppal) [reply smoke-test.go]: moved to a global variable so each method can be use it. const cause issues trying to take address of when trying to pass in a pointer to it.

**Review by craigbr — APPROVED** [review]

**Pull request merged**